src/sequences.ts

```diff
@@ -69,19 +72,27 @@ export default class SequenceService {
       meta.last = i + 1 === sequenceLength
       const assertion = sequence.assertions[i]
       sequence = Object.assign(sequence, meta)
-      // eslint-disable-next-line no-await-in-loop
+
       indentLog(6, `Assert: ${assertion.description} (type: ${assertion.type})`)
 
       const Asserter = AsserterLookup[assertion.type]
-      const asserter = new Asserter({
-        assertion,
-        repoFullName,
-        dryRun,
-        branchName,
-        templateDir: sequence.templateDir,
-      })
-      // eslint-disable-next-line no-await-in-loop
-      await asserter.run()
+
+      try {
+        if (!Asserter) throw new Error(`Invalid asserter type ${assertion.type}`)
+
+        const asserter = new Asserter({
+          assertion,
+          repoFullName,
+          dryRun,
+          branchName,
+          templateDir: sequence.templateDir,
+        })
+        await asserter.run()
+      } catch (error) {
+        await exec(`git -C ${workingDir} checkout ${masterMain}`)
```

Comment:
This is not well tested. I just know you won't be able to run the workflow again without cleaning up if an error happens because the branch is already created, and you can't run `git pull` because the origin isn't set up. However, what happens if there are uncommitted files? I'm sure there are other scenarios too.